RijsMuseum/app/src/main/java/com/example/rijsmuseum/MainActivity.kt

```diff
@@ -1,11 +1,64 @@
 package com.example.rijsmuseum
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import android.widget.FrameLayout
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.network.getters.CollectionsGetter
+import com.example.network.models.Collections
+import com.example.network.models.CollectionsDetails
+import com.example.rijsmuseum.adapter.RecyclerAdapter
+import com.example.rijsmuseum.fragments.DetailsFragment
+import com.example.rijsmuseum.viewmodel.MainViewModel
+
+class MainActivity : AppCompatActivity(), RecyclerAdapter.OnItemClickListener {
+    private lateinit var getRequestButton: Button
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var flFragment: FrameLayout
+
+    private lateinit var viewModel: MainViewModel
+
+    private lateinit var adapter: RecyclerAdapter
 
-class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
+
+        recyclerView = findViewById(R.id.recyclerView)
+        flFragment = findViewById(R.id.flFragmentDetails)
+
+        adapterSettings()
+
+        //Button to get the list of artifacts
+        getRequestButton = findViewById(R.id.getButton)
+        getRequestButton.setOnClickListener {
+            viewModel.getCollectionsRequest(object : CollectionsGetter.DataReadyCallback {
+                override fun onDataReady(data: List<Collections.ArtObject>) {
+                    adapter.updateCollectionsData(data)
+                }
+            })
+        }
+    }
+
+    private fun adapterSettings() {
+        adapter = RecyclerAdapter()
+        recyclerView.layoutManager = LinearLayoutManager(this)
+        recyclerView.adapter = adapter
+        adapter.setOnItemClickListener(this)
+    }
+
+    override fun onItemClick(cDetailsArtObject: CollectionsDetails.ArtObject, position: Int) {
```

Comment:
ok, what's the deal to send both the CollectionsDetails.ArtObject and the position? :)